src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -859,7 +896,20 @@ private SupportedPlatform GetCurrentRunningOS()
                 return context;
             }
 
+            // If computeHashes exists and we already added in the cache,
+            // we can just return the context.
+            if (_computeHashes && _analysisLoggerCache.ContainsKey(context.Hashes.Sha256))
```

Comment:
>_analysisLoggerCache [](http://example.com/codeflow?start=34&length=20)

Why is it correct here not to call into the logger? How are the cached results returned in this case?

Reply:
when this happen, we will return the context and mark as completed.

From that, the thread that is writing will look in the cache for the an AnalysisLogger with that hash.

With that, it would clone the results and log it.